File: Searching and Sorting Algorithms/Bubble Sort/bubbleSort.py
def print_array(array):
    for i in array:
        print(i, end=" ")
    print("")


def bubble_sort(array):
    for i in range(len(array)):
        swapped = False

        for j in range(0, len(array)-i-1):
            if array[j] >= array[j+1]:
                tmp = array[j+1]
                array[j+1] = array[j]
                array[j] = tmp
                swapped = True

        if not swapped:
            break


foo_array = [9, 4, 3, 5, 1]
print_array(foo_array)
bubble_sort(foo_array)
print_array(foo_array)

File: Searching and Sorting Algorithms/Bubble Sort/bubbleSort.cpp
#include <array>
#include <iostream>

template <typename TYPE, size_t SIZE>
void print_array(std::array<TYPE, SIZE> array) {
  for (const auto& i : array) {
    std::cout << i << " ";
  }
  std::cout << "\n";
}

template <typename TYPE, size_t SIZE>
void bubble_sort(std::array<TYPE, SIZE>& array) {
  for (size_t i = 0; i < SIZE; i++) {
    bool swapped = false;
    for (size_t j = 0; j < SIZE - i - 1; j++) {
      if (array[j] > array[j + 1]) {
        TYPE temp = array[j + 1];
        array[j + 1] = array[j];
        array[j] = temp;
        swapped = true;
      }
    }
    if (!swapped)
      break;
  }
}

int main() {
  std::array<int, 7> foo_array = {6, 4, 9, 4, 3, 5, 1};
  print_array(foo_array);
  bubble_sort(foo_array);
  print_array(foo_array);
}